datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala

```diff
@@ -0,0 +1,181 @@
+package za.co.absa.hermes.datasetComparison
+
+import com.typesafe.config.{Config, ConfigFactory}
+import org.apache.spark.sql._
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types.{StructField, StructType}
+import za.co.absa.hermes.utils.HelperFunctions
+
+object DatasetComparisonJob {
+  private val conf: Config = ConfigFactory.load
```

Comment:
```suggestion
  private val conf: Config = ConfigFactory.load()
```
Not eaxctly a side-effect free getter (performs IO to read in `.conf` file), so it should have parens.